chips/rp2040/src/pio.rs

```diff
@@ -0,0 +1,1233 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OxidOS Automotive 2024.
+//
+// Author: Radu Matei <radu.matei.05.21@gmail.com>
+
+//! Programmable Input Output (PIO) hardware.
+
+use kernel::debug;
+use kernel::utilities::registers::interfaces::{ReadWriteable, Readable, Writeable};
+use kernel::utilities::registers::{register_bitfields, register_structs, ReadOnly, ReadWrite};
+use kernel::utilities::StaticRef;
+
+use crate::gpio::{GpioFunction, RPGpio, RPGpioPin};
+
+const NUMBER_STATE_MACHINES: usize = 4;
+const NUMBER_INSTR_MEMORY_LOCATIONS: usize = 32;
+
+#[repr(C)]
+struct InstrMem {
+    // Write-only access to instruction memory locations 0-31
+    instr_mem: ReadWrite<u32, INSTR_MEMx::Register>,
```

Comment:
Tock registers does have support for a `WriteOnly` type -- seems like maybe that is what you want here given the comment?